integration/common/openlineage/common/provider/dbt.py

```diff
@@ -627,15 +664,34 @@ def extract_job_namespace(self, profile: Dict):
 
     def extract_namespace(self, profile: Dict) -> str:
         """Extract namespace from profile's type"""
-        if profile['type'] == 'snowflake':
+        if self.adapter_type == Adapter.SNOWFLAKE:
             return f"snowflake://{profile['account']}"
-        elif profile['type'] == 'bigquery':
+        elif self.adapter_type == Adapter.BIGQUERY:
             return "bigquery"
-        elif profile['type'] == 'redshift':
+        elif self.adapter_type == Adapter.REDSHIFT:
             return f"redshift://{profile['host']}:{profile['port']}"
+        elif self.adapter_type == Adapter.SPARK:
+            port = ""
+
+            if 'port' in profile:
+                port = f":{profile['port']}"
+            elif profile['method'] in [
+                SparkConnectionMethod.HTTP.value, SparkConnectionMethod.ODBC.value
+            ]:
+                port = '443'
+            elif profile['method'] == SparkConnectionMethod.THRIFT.value:
+                port = '10001'
+
+            if profile['method'] in SparkConnectionMethod.methods():
+                return f"spark://{profile['host']}{port}"
+            else:
+                raise NotImplementedError(
+                    f"Connection method `{profile['method']}` is not "
+                    f"supported for spark adapter."
+                )
```

Comment:
I'd invert the if condition and move it up to exit early if the `profile['method']` isn't supported.